test/test-helpers/did-render.ts

```diff
@@ -0,0 +1,13 @@
+import Application from '../../src/application';
+
+async function didRender(app: Application): Promise<void> {
+  return new Promise<void>(resolve => {
+    let watcher = setInterval(function() {
+      if (app['_scheduled']) return;
```

Comment:
Just realized that `scheduled` is set to `false` _before` `this._rerender()` is called. What if we did this instead?

```typescript
scheduleRerender(): void {
  if (this._scheduled || !this._rendered) return;

  this._scheduled = true;
  requestAnimationFrame(() => {
    this._rerender();
    this._scheduled = false;
  });
}
```

The advantage of my approach here is that it's just an internal test helper. In the future, if/when the situation changes, we can change the helper to key off of whatever we need it to. There's no public API to worry about.